src/query/functions/src/scalars/interval.rs

```diff
@@ -79,3 +86,227 @@ fn register_interval_to_string(registry: &mut FunctionRegistry) {
         ),
     );
 }
+
+fn register_interval_add_sub(registry: &mut FunctionRegistry) {
+    registry.register_passthrough_nullable_2_arg::<IntervalType, IntervalType, IntervalType, _, _>(
+        "plus",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_2_arg::<IntervalType, IntervalType, IntervalType>(
+            |a, b, output, _| {
+                output.push(months_days_micros::new(
+                    a.months() + b.months(),
+                    a.days() + b.days(),
+                    a.microseconds() + b.microseconds(),
+                ))
+            },
+        ),
+    );
+
+    registry
+        .register_passthrough_nullable_2_arg::<TimestampType, IntervalType, TimestampType, _, _>(
+            "plus",
+            |_, _, _| FunctionDomain::MayThrow,
+            vectorize_with_builder_2_arg::<TimestampType, IntervalType, TimestampType>(
+                |a, b, output, ctx| {
+                    // plus microseconds and days
+                    let ts = a
+                        .wrapping_add(b.microseconds())
```

Comment:
why not use `total_micros `

Reply:
Add total micros is not right. 

If interval compression 

30day is equal with 1 month.

But if timestamp add 30 days and add 1month maybe diff.